src/main/java/kitty/Ui.java

```diff
@@ -0,0 +1,127 @@
+package kitty;
+
+import kitty.kittyexceptions.FindException;
+import kitty.kittyexceptions.MarksException;
+import kitty.tasks.Task;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Ui {
+    private static final String DIVISION_LINE = "--------------------------";
+    private boolean isWorking;
+    private TaskList tasks;
+    private Storage storage;
+
+    public Ui(TaskList tasks, Storage storage) {
+        isWorking = true;
+        this.tasks = tasks;
+        this.storage = storage;
+    }
+
+    public void run() {
+        showGreetingMessage();
+
+        String command = "";
+        Scanner sc = new Scanner(System.in);
+        while (isWorking) {
+            command = sc.nextLine();
+            if (command.contains("bye")) {
+                isWorking = false;
+                showGoodbyeMessage();
+                continue;
+            }
+            try {
+                Parser.parseFirstWord(command, this, tasks, storage);
+            } catch (FindException e) {
+                showErrorMessage(e.toString());
+            } catch (MarksException e) {
+                showErrorMessage(e.toString());
+            } catch (NumberFormatException e) {
+                showErrorMessage("Please Input valid number.\n");
+            } catch (IndexOutOfBoundsException e) {
+                showErrorMessage("Index out of bound!!! Please input number within the range of list size.\n");
+            }
+        }
+    }
+
+    public void showGreetingMessage() {
```

Comment:
I noticed that showGreetingMessage() and showGoodbyeMessage() have similar code for displaying messages and division lines. You could simplify things by creating a common method for these repetitive parts. It could make your code cleaner and easier to manage!